lib/devise/strategies/passwordless_authenticatable.rb

```diff
@@ -18,7 +18,16 @@ def valid?
 
       def authenticate!
         if params[:user].present?
-          user = User.find_by(email: params[:user][:email])
+          email = params.dig(:user, :email)
+
+          identity = ParticipantIdentity.find_by(email: email)
+          if identity.present?
+            confirmed_email = identity.email
```

Comment:
Why do we need `confirmed_email`? Won't it always just be the same as `email`, or am I missing something?

Reply:
The supplied email could be assorted case, there was an existing spec that tested the email used in the sign in email was the "normalized" version